src/lib/one-graph/graphql-helpers.js

```diff
@@ -0,0 +1,533 @@
+const dotProp = require('dot-prop')
+const {
+  TypeInfo,
+  getNamedType,
+  isEnumType,
+  isInputObjectType,
+  isInterfaceType,
+  isListType,
+  isNonNullType,
+  isObjectType,
+  isUnionType,
+  isWrappingType,
+  parseType,
+  print,
+  typeFromAST,
+  visit,
+  visitWithTypeInfo,
+} = require('graphql')
+
+const capitalizeFirstLetter = (string) => string.charAt(0).toUpperCase() + string.slice(1)
+
+const gatherAllReferencedTypes = (schema, query) => {
+  const types = new Set([])
+  const typeInfo = new TypeInfo(schema)
+  visit(
+    query,
+    visitWithTypeInfo(typeInfo, {
+      enter: () => {
+        const fullType = typeInfo.getType()
+        if (fullType) {
+          const typ = getNamedType(fullType)
+          if (typ) types.add(typ.name.toLocaleLowerCase().replace('oneme', ''))
+        }
+      },
+    }),
+  )
+  const result = [...types]
+  return result
+}
+
+const extractVariableNameStringPair = (varDef) => [varDef.variable.name.value, print(varDef.type)]
+
+const gatherVariableDefinitions = (definition) => {
+  const varDefs = dotProp.get(definition, 'variableDefinitions.') || []
+  return varDefs.map(extractVariableNameStringPair).sort(([left], [right]) => left.localeCompare(right))
+}
+const typeScriptForGraphQLType = (schema, gqlType) => {
+  const scalarMap = {
+    String: 'string',
+    ID: 'string',
+    Int: 'number',
+    Float: 'number',
+    Boolean: 'boolean',
+  }
+  if (isListType(gqlType)) {
+    const subType = typeScriptForGraphQLType(schema, gqlType.ofType)
+    return `Array<${subType}>`
+  }
+  if (isObjectType(gqlType) || isInputObjectType(gqlType)) {
+    const fields = Object.values(gqlType.getFields()).map((field) => {
+      const nullable = !isNonNullType(field.type)
+      const type = typeScriptForGraphQLType(schema, field.type)
+      const description = field.description
+        ? `/**
```

Comment:
`outdent` would make this easier to read. I think we already have this in the repo.